internal/server/server.go

```diff
@@ -220,3 +220,87 @@ func VerifyServerStarted(startRequest request.StartServerRequest, meta interface
 	}
 	return nil
 }
+
+// RemoveOldServerTags removes tags from server.
+// deletes the tag if no server left with the tag name
```

Comment:
Are tags *only* related to servers (for now?) Seems a bit dangerous to me to auto-delete but not sure I understand the semantics/issues involved.

Reply:
yes the way tags are implemented for now are only related to server, that's why

Reply:
:+1: Got it, makes sense, then. I'm  a bit concerned over what happens when/if they don't? Is there a downside to not deleting the tags? Do we have a way of querying the resources/servers of a tag - if we do and that seems like a generic way that should work in the future as well, could we use it to make sure? 

If it seems too complex/impossible to do now, maybe at the very least move the deletion to a separate method and make it explicit that this is the functionality now? Otherwise, figuring it out would need one to look into RemoveOldServerTags which doesn't really give an indication that it will also delete tags.

Reply:
removing the tags when there are no server on them is the best way of keeping them clean, I don't know that is the limit of tags resource in the API, but I think this is the best way to do it since there are also tag resources in tf that may conflict if the tags were deleted from server and wanted to be used as a standalone tag resource.